Makefile

```diff
@@ -153,15 +154,32 @@ root/.built: \
 	( \
 		cd root/lib/python$(PYMINOR); \
 		rm -fr `cat ../../../remove_modules.txt`; \
-		rm encodings/mac_*.py; \
 		rm -fr test; \
 		find . -name "*.wasm.pre" -type f -delete ; \
 		find -type d -name __pycache__ -prune -exec rm -rf {} \; \
 	)
 	touch root/.built
 
 
-$(CPYTHONLIB): emsdk/emsdk/.complete
+ccache/emcc:
+	if hash ccache &>/dev/null; then \
+		mkdir -p $(PYODIDE_ROOT)/ccache ; \
+    ln -s `which ccache` $(PYODIDE_ROOT)/ccache/emcc ; \
+  else \
+    ln -s emsdk/emsdk/emscripten/tag-1.38.4/emcc $(PYODIDE_ROOT)/ccache/emcc; \
+  fi
+
+
+ccache/em++:
+	if hash ccache &>/dev/null; then \
+		mkdir -p $(PYODIDE_ROOT)/ccache ; \
+    ln -s `which ccache` $(PYODIDE_ROOT)/ccache/em++ ; \
```

Comment:
maybe also set a smaller max cache size limit? e.g.
```
ccache -M 512M
```
by default it's 5 GB.

Reply:
Why?  I can't find what the maximum cache size is on Circle-CI, but I see no reason not to use the maximum we have available to us.

Reply:
I guess if it's large it still take some time to restore and it probably means that we are keeping outdated files in the cache. [Scipy does this](https://github.com/iodide-project/pyodide/pull/88#issuecomment-407480885). But you are right, we could always add it later if it ever becomes a problem.